src/System.Text.Http/System/Text/Http/Parser/HttpParser.cs

```diff
@@ -107,271 +105,151 @@ public HttpParser(bool showErrorDetails)
                 }
+                query = query.Slice(0, spaceIndex);
             }
 
-            // End of query string not found
-            if (offset == length)
-            {
-                RejectRequestLine(data, length);
-            }
+            target = target.Slice(0, path.Length + query.Length);
 
-            var targetBuffer = new ReadOnlySpan<byte>(data + pathStart, offset - pathStart);
-            var query = new ReadOnlySpan<byte>(data + queryStart, offset - queryStart);
+            // Version
 
-            // Consume space
-            offset++;
+            // Skip space
+            ReadOnlySpan<byte> version = data.Slice(offset + target.Length + 1);
 
-            // Version
-            var httpVersion = HttpUtilities.GetKnownVersion(data + offset, length - offset);
-            if (httpVersion == Http.Version.Unknown)
+            if (!HttpUtilities.GetKnownVersion(version, out Http.Version httpVersion))
             {
-                if (data[offset] == ByteCR || data[length - 2] != ByteCR)
+                if (version[0] == ByteCR || data[data.Length - 2] != ByteCR)
                 {
                     // If missing delimiter or CR before LF, reject and log entire line
-                    RejectRequestLine(data, length);
+                    RejectRequestLine(data);
                 }
                 else
                 {
                     // else inform HTTP version is unsupported.
-                    RejectUnknownVersion(data + offset, length - offset - 2);
+                    RejectUnknownVersion(data.Slice(offset, data.Length - offset - 2));
                 }
             }
 
             // After version's 8 bytes and CR, expect LF
-            if (data[offset + 8 + 1] != ByteLF)
+            if (version[8 + 1] != ByteLF)
             {
-                RejectRequestLine(data, length);
+                RejectRequestLine(data);
             }
 
-            var line = new ReadOnlySpan<byte>(data, length);
-
-            handler.OnStartLine(method, httpVersion, targetBuffer, pathBuffer, query, customMethod, pathEncoded);
+            handler.OnStartLine(method, httpVersion, target, path, query, customMethod, pathEncoded);
         }
 
-        public unsafe bool ParseHeaders<T>(T handler, in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined, out int consumedBytes) where T : IHttpHeadersHandler
+        public bool ParseHeaders<T>(
+            T handler,
+            in ReadOnlySequence<byte> buffer,
+            out SequencePosition consumed,
+            out SequencePosition examined,
+            out int consumedBytes) where T : IHttpHeadersHandler
         {
             consumed = buffer.Start;
             examined = buffer.End;
             consumedBytes = 0;
 
-            var bufferEnd = buffer.End;
+            BufferReader<byte> reader = new BufferReader<byte>(buffer);
+            ReadOnlySpan<byte> eol = s_Eol;
```

Comment:
Update the static, constant, byte arrays to be ReadOnlySpan properties.

So, s_Eol becomes:
```C#
ReadOnlySpan<byte> Eol => new byte [] {(byte)'\r', (byte)'\n'};
```